### ISSUE TITLE
broken on 1.58

### ISSUE THREAD

**Issue by HexCodeFFF** [afonsosousah/gtaonlinespeedometer#4]
title

playerspeed and xyz work but it cant detect when you get in a vehicle

**Comment by afonsosousah** [comment]
I haven't updated it yet, the memory offsets probably have changed
Maybe some time i will remake the app and add some things... I hope I have time xD

**Comment by HexCodeFFF** [comment]
Yeah I get that

You don't need to rewrite the whole thing, I'd just update the memory offsets and release a patch like you did for 1.52 :p (if you want to ofc)

I think the offset for "player is in vehicle" changed and I know that Kiddion knows this offset for his menu so maybe ask him?

**Comment by afonsosousah** [comment]
if i didn't forget how to do it, i can get the offsets, its not that hard since its a simple value

i'll work on it if and when i have time, thank you for writing

**Comment by Kim123LP** [comment]
@afonsosousah Can you either update to 1.58 as soon as possible or provide the source code? I don't think that once you know how to find addresses it can be that hard. And if you don't feel like it anymore, at least let's do it ourselves.

**Comment by HexCodeFFF** [comment]
i think kim phrased it a little rude but the source code would be cool

**Comment by afonsosousah** [comment]
i uploaded the source code to the folder named "GTA+Info"

i don't why i didn't do this earlier xd

well if you have any question about the source just create a new issue and i will answer when i can

**Comment by actopozipc** [comment]
I had a look in the code. 
1. a reference was missing for some reason. If anyone wants to edit this, you probably need to add the CircularGauge.dll as reference path.
2. Since I dont understand how the offset value gets calculated, I just changed the menu to true, so the speedometer gets shown all the time. Its not a clean solution but a working one. Edit: You also need to edit the getter of the IsInVehicle property to always return true
3. In the name of jesus Christ, have you ever heard about comments?

**Comment by afonsosousah** [comment]
yeah, my code isn't pretty at all, and that is code i wrote as i was learning c#, so its even worse xd
